### ISSUE TITLE
下载弹窗导致程序崩溃

### ISSUE THREAD

**Issue by qq06314488** [MicrosoftEdge/WebView2Feedback#1765]
**Description**

当我尝试从webview2中下载资源文件，会在右上角弹出下载对话框，当我不关闭这个对话框直接关闭网页窗口的时候，会导致整个程序的崩溃

System.InvalidOperationException
  HResult=0x80131509
  Message=CoreWebView2 members cannot be accessed after the WebView2 control is disposed.
  Source=Microsoft.Web.WebView2.Core
  StackTrace:
   在 Microsoft.Web.WebView2.Core.CoreWebView2Controller.MoveFocus(CoreWebView2MoveFocusReason reason)
   在 Microsoft.Web.WebView2.Wpf.WebView2.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   在 System.Windows.Interop.HwndHost.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   在 MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   在 System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   在 System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   在 System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   在 MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)

  此异常最初是在此调用堆栈中引发的: 
    [外部代码]

内部异常 1:
COMException: 组或资源的状态不是执行请求操作的正确状态。 (0x8007139F)

**Version**
WebView2，版本：1.0.1018-prerelease
SDK: .NET5.0
Runtime: 
Framework: WPF
OS: Win10

**Repro Steps**

1：启动程序后，进入主窗口A，然后通过A窗口进入窗口B（这会触发webView.CoreWebView2.NewWindowRequested事件，从新建一个B窗口）
2：在B窗口中点击下载资源的选项，弹出下载对话框
3：在不关闭下载对话框的情况下，直接关闭B窗口，会导致主窗口A的崩溃。

![Snipaste_2021-09-25_22-01-08](https://user-images.githubusercontent.com/22559946/134774198-94e2c098-250f-4379-a3f5-23ae688ccd7f.png)

如果不调用webView.Dispose();释放资源就不会发生上面崩溃的情况，但是会导致资源的泄漏。

**Screenshots**

**Additional context**

`    public partial class MainWindow : Window

    {
        public MainWindow()
        {
            InitializeComponent();
            webView.Source = new Uri("https://im.qq.com");
            InitWebView();
        }

        private async void InitWebView()
        {
            await webView.EnsureCoreWebView2Async();
            webView.CoreWebView2.NewWindowRequested += CoreWebView2_NewWindowRequested;
        }

        private void CoreWebView2_NewWindowRequested(object? sender, Microsoft.Web.WebView2.Core.CoreWebView2NewWindowRequestedEventArgs e)
        {
            e.Handled = true;
            MainWindow main = new MainWindow();
            main.webView.Source = new Uri(e.Uri);
            main.Show();
        }
        protected override void OnClosed(EventArgs e)
        {
            // 释放资源
            //webView.Dispose();
            base.OnClosed(e);
        }
    }
`

**Comment by qq06314488** [comment]
Webview2 - Microsoft Edge Developer






------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Nic Champagne Williamson ***@***.***&gt;; 
发送时间: 2021年9月28日(星期二) 上午8:56
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [MicrosoftEdge/WebView2Feedback] 下载弹窗导致程序崩溃 (#1765)





 
Hey @qq06314488 - This looks like the same crash as #1723. What runtime version are you seeing this on?
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by qq06314488** [comment]
Webview2 - Microsoft Edge Developer







------------------ 原始邮件 ------------------
发件人: "Nic Champagne Williamson ***@***.***&gt;; 
发送时间: 2021年9月28日(星期二) 上午8:56
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [MicrosoftEdge/WebView2Feedback] 下载弹窗导致程序崩溃 (#1765)





 
Hey @qq06314488 - This looks like the same crash as #1723. What runtime version are you seeing this on?
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by qq06314488** [comment]
> Hey @qq06314488 - This looks like the same crash as #1723. What runtime version are you seeing this on?

您好，我的
Edge版本 94.0.992.31 (官方内部版本) (64 位）

**Comment by champnic** [comment]
When you say "Edge version (Edge版本)" do you mean the browser or the WebView2 Runtime?
![image](https://user-images.githubusercontent.com/45700886/135140330-b9f94df8-2074-4c44-887d-26bbb8854e85.png)

**Comment by qq06314488** [comment]
> When you say "Edge version (Edge版本)" do you mean the browser or the WebView2 Runtime? ![image](https://user-images.githubusercontent.com/45700886/135140330-b9f94df8-2074-4c44-887d-26bbb8854e85.png)
![1123](https://user-images.githubusercontent.com/22559946/135200710-4482c70a-97a2-494c-9465-8f56c2e7aa85.png)
我的版本和您一样的。// My version is the same as yours.

**Comment by champnic** [comment]
Thanks for confirming. That version should have the expected fix, so we'll take a look and see if this might be something else. I've added a bug on our backlog.

**Comment by champnic** [comment]
We were unable to reproduce the issue using the code you provided and the latest stable runtime. Can you try the scenario again and verify your app is using 94.0.992.31?

**Comment by qq06314488** [comment]
> We were unable to reproduce the issue using the code you provided and the latest stable runtime. Can you try the scenario again and verify your app is using 94.0.992.31?
很抱歉现在才回复您。
我确认我的版本是94.0.992.31，我录制了录像，您可以看一下
第一次崩溃，是因为我没有关闭下载弹窗，直接关闭了下载网页的窗口，导致父窗口崩溃
第二次可以正常运行，是因为我先关闭了下载弹窗，然后关闭下载页面窗口，所以父窗口是正常的
项目文件：https://github.com/qq06314488/WebApp
录制的视频文件：https://alywp.net/44BLQW

**Comment by Deepanghoshs** [comment]
I am also facing the same issue mentioned by @qq06314488 , I have mentioned the same  in the bug 
[1734](https://github.com/MicrosoftEdge/WebView2Feedback/issues/1734)
[1723](https://github.com/MicrosoftEdge/WebView2Feedback/issues/1723)

**Comment by Deepanghoshs** [comment]
@champnic  My Environment Details
Windows 10 pro with latest updates Version 10.0.19043 Build 19043
MS Edge WEBview2 Runtime 94.0.992.31
Microsoft Edge Version 96.0.1028.0 (Official build) dev (64-bit)

**Comment by Deepanghoshs** [comment]
is any update on crashing issue?

**Comment by qq06314488** [comment]
> We just put in a fix for this in runtimes 97.0.1060.0+. Can you test against the latest Canary and see if the issue is resolved for you? https://docs.microsoft.com/en-us/microsoft-edge/webview2/how-to/set-preview-channel
Yes, the Bug has been fixed。latest Canary Edge It works correctly

**Comment by champnic** [comment]
Thanks for confirming! We are also going to put this change into the 96 version soon.
